middleware/AuthorizationHandler.md

```diff
@@ -13,15 +12,22 @@ A piece of client-side middleware intended to allow the use of an authorization
 - Certain workloads error out when an unexpected header is present in the request. The middlewares should check:
   - If the request URL is a Graph serve endpoint or a custom url provided by the developer, then append or update the authorization header.
   - Else the middleware should delete authorization header.
- 
 
 Take a request object and use authorization provider to add Authorization header.  
 
+### Additional requirements for Kiota clients
+
+- MUST NOT attempt to update the authorization header if it is already present (unless we ran into an authenticate challenge). This is to ensure customers can use the same http client for arbitrary requests and with a generated client/request adapter without running into double authentication issues.
+- MUST accept a [BaseBearerAuthenticationProvider](https://github.com/microsoft/kiota-dotnet/blob/main/src/abstractions/authentication/BaseBearerTokenAuthenticationProvider.cs) and use the associated AccessTokenProvider to obtain the access token. This is to ensure the client initialization remains simple.
```

Comment:
Since we are in the http middleware at this point, we will be working with `HttpRequestMessage` instances (in dotnet) while the `BaseBearerAuthenticationProvider` works on `RequestInformation` instances, 

To prevent round tripping of conversion for the authprovider to do its job, should we only accept the AccessTokenProvider instead?

Reply:
the access token provider takes a URI as an argument and returns the token as a result. It's the authentication provider that takes the request information. Here we'd simply be using the authentication provider to get the access token provider.
Maybe there's a way i can update the wording to make the distinction clearer?

Reply:
added the example to the spec

Reply:
Thanks for adding the example here. Resolving this for now.